src/NodeServices/Attributes.cs

```diff
@@ -342,4 +342,26 @@ public class KeepReferenceThisAttribute : Attribute
     public class IsLacingDisabledAttribute : Attribute
     {
     }
+
+    /// <summary>
+    /// This attribute is applied to a function that has one or more parameters
+    /// as lists. It can be used to control arguments to the function
+    /// from being promoted to arrays or arrays of higher dimension when the VM tries
+    /// to do method resolution and match argument(s) to the function parameter(s). 
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AllowArrayPromotionAttribute : Attribute
+    {
+        public bool IsAllowed { get; private set; }
+
+        public AllowArrayPromotionAttribute()
+        {
+            IsAllowed = true;
```

Comment:
so it looks like the default here was false previously? And now it's true? Am I interpreting that correctly? If so - does that have the potential for lots of behavior changes?

Reply:
Yup or should I add an attribute called `DisallowArrayPromotion` in which case we can get rid of the extra boolean flag.